### PR TITLE
Simple delay

### PR DESCRIPTION
**Pull request by gusy**

### CODE REVIEW

File: lib/couch_tap/query_executor.rb [hunk 1]
```diff
@@ -110,7 +112,8 @@ def run_transaction(seq)
           end
 
           logger.debug "Changes applied, updating sequence number now to #{@seq}"
-          update_sequence(seq)
+          @metrics.gauge('delay', (Time.now - (@buffer.newest_updated_at || Time.now)).round)
```
Comment (calonso) [inline lib/couch_tap/query_executor.rb]: I don't think we can assume we're fully updated if no `newest_updated_at` could be calculated. I'd rather send nothing. Specially given that below, in `update_sequence` we're saving `nil` in that case

File: test/unit/query_executor_test.rb [hunk 2]
```diff
@@ -278,14 +278,19 @@ def test_timer_signal_is_skipped_if_last_transaction_ran_too_close
   end
 
   def test_empty_batches_are_skipped
+    Timecop.freeze
```
Comment (calonso) [inline test/unit/query_executor_test.rb]: This is a different test. Something like `test_empty_batches_update_last_transaction_at`.

File: test/functional/functional_changes_test.rb [hunk 3]
```diff
@@ -40,22 +40,25 @@ def test_insert_multiple_sales
...
-    assert_includes sales, sale_id: "11", code: "Code 2", amount: 1000
-    assert_includes sales, sale_id: "12", code: "Code 3", amount: 325
+    assert_includes sales, sale_id: "10", code: "Code 1", amount: 600, updated_at: nil
+    assert_includes sales, sale_id: "11", code: "Code 2", amount: 1000, updated_at: nil
+    assert_includes sales, sale_id: "12", code: "Code 3", amount: 325, updated_at: nil
     assert_sequence changes.seq, 125
   end
 
   def test_insert_multiple_sales_in_same_batch
+    Timecop.freeze(Time.now)
```
Comment (calonso) [inline test/functional/functional_changes_test.rb]: No need to freeze at this level. Just set fixed times for `updated_at` values and assert them. Given that all records have the field, `Time.now` shouldn't be used, right?

File: test/functional/functional_changes_test.rb [hunk 4]
```diff
@@ -70,12 +70,11 @@ def test_insert_multiple_sales_in_same_batch
 
     sales = @database[:sales].to_a
     assert_equal 3, sales.count
-    assert_includes sales, sale_id: "10", code: "Code 1", amount: 600, updated_at: (Time.now - 120).round
+    assert_includes sales, sale_id: "10", code: "Code 1", amount: 600, updated_at: (example_time - 120).round
```
Comment (calonso) [inline test/functional/functional_changes_test.rb]: Why do you need to `.round`?

File: test/unit/query_buffer_test.rb [hunk 5]
```diff
@@ -101,6 +101,14 @@ def test_delete_increases_size
     assert_equal 2, buffer.delete(item_to_delete(987))
   end
 
+   def test_clear_deletes_newest_updated_at
+    buffer = CouchTap::QueryBuffer.new
+    assert_equal 1, buffer.insert(item_to_insert(true, 123, { updated_at: Time.now.rfc2822 }))
+    assert_equal 2, buffer.delete(item_to_delete(987))
+    buffer.clear
```
Comment (calonso) [inline test/unit/query_buffer_test.rb]: assert that `newest_updated_at` had something before clearing

File: test/unit/query_buffer_test.rb [hunk 6]
```diff
@@ -138,6 +146,23 @@ def test_can_iterate_over_entities
     assert_equal %i(item another), entity_names
   end
 
+  def test_newest_updated_at
+    dummy_date = Time.new(2008,6,21, 13,30,0)
+    buffer = CouchTap::QueryBuffer.new
+
+    buffer.insert(item_to_insert(true, 123, { updated_at: dummy_date.rfc2822 }))
+    assert_equal buffer.newest_updated_at, dummy_date
```
Comment (calonso) [inline test/unit/query_buffer_test.rb]: Swap the arguments. The signature is `assert_equal expected, found`. It makes a difference when reading failures.

File: test/unit/query_buffer_test.rb [hunk 7]
```diff
@@ -138,6 +146,23 @@ def test_can_iterate_over_entities
...
+
+    buffer.insert(item_to_insert(true, 456))
+    assert_equal buffer.newest_updated_at, dummy_date
```
Comment (calonso) [inline test/unit/query_buffer_test.rb]: Swap args

File: test/unit/query_buffer_test.rb [hunk 8]
```diff
@@ -138,6 +146,23 @@ def test_can_iterate_over_entities
...
+
+    buffer.insert(item_to_insert(true, 123, { updated_at: (dummy_date - 10).rfc2822 }))
+    assert_equal buffer.newest_updated_at, dummy_date
```
Comment (calonso) [inline test/unit/query_buffer_test.rb]: Swap args

File: test/unit/query_buffer_test.rb [hunk 9]
```diff
@@ -138,6 +146,23 @@ def test_can_iterate_over_entities
...
+
+    buffer.insert(item_to_insert(true, 123, { updated_at: (dummy_date + 10).rfc2822 }))
+    assert_equal buffer.newest_updated_at, (dummy_date + 10)
```
Comment (calonso) [inline test/unit/query_buffer_test.rb]: Swap args

File: test/unit/query_executor_test.rb [hunk 10]
```diff
@@ -120,6 +122,7 @@ def test_create_and_delete_same_row
 
     assert_equal 0, executor.database[:items].where(item_id: 123).count
     assert_equal 1, executor.database[:couch_sequence].where(name: 'items').first[:seq]
+    assert_equal DUMMY_DATE, executor.database[:couch_sequence].where(name: 'items').first[:last_transaction_at]
```
Comment (calonso) [inline test/unit/query_executor_test.rb]: `DUMMY _DATE` is only read here. Don't need to be a constant. Use a local variable instead.

File: test/functional/functional_changes_test.rb (hunk 4 above)
Comment (gusy) [inline test/functional/functional_changes_test.rb]: In this I don't, but some others I need so that string conversion does not look precision

Comment (calonso) [inline test/functional/functional_changes_test.rb]: Now that you've `frozen` and saved the object I don't think you need it. Double check it, though.

**Pull request merged**